pkg/logs/decoder/auto_multiline_handler.go

```diff
@@ -48,15 +48,14 @@ func (d *DetectedPattern) Get() *regexp.Regexp {
 type AutoMultilineHandler struct {
 	multiLineHandler  *MultiLineHandler
 	singleLineHandler *SingleLineHandler
-	inputChan         chan *Message
-	outputChan        chan *Message
+	outputFn          func(*Message)
```

Comment:
Is this to hide the channel or maybe it's this only for unit tests purpose? (or maybe you were doing more in these func previous to the push?)
It feels a bit weird to have the AutoMultilineHandler or the `SingleLineHandler` using a channel created in `NewDecoderWithFraming` without nobody really "owning" it (for instance we can never close it). Not a big deal but it looks weird to me.

Reply:
I'm not sure what you're asking -- the outputFn is how the line handler produces its output, so it's pretty important!  And, it doesn't have a channel here.  In a decoder, since this is the last component, outputFn will write to decoder.OuptutChan, but that is very much owned by the decoder and will be closed when its input is closed.

Reply:
Also, note that this is part of #11095, so perhaps better to review there.